source/vulkan/vulkan_hooks.cpp

```diff
@@ -454,14 +693,399 @@ VkResult VKAPI_CALL vkQueuePresentKHR(VkQueue queue, const VkPresentInfoKHR *pPr
+{
+	PFN_vkCmdClearDepthStencilImage trampoline;
+
+	{ const std::lock_guard<std::mutex> lock(s_mutex);
+		trampoline = s_device_dispatch.at(get_dispatch_key(commandBuffer)).CmdClearDepthStencilImage;
+		assert(trampoline != nullptr);
+	}
+
+	// no device associated (this cannot happen normally)
+	if (const auto it = s_command_buffer_mapping.find(commandBuffer); it != s_command_buffer_mapping.end())
+	{
+		VkDevice device = s_command_buffer_mapping.at(commandBuffer);
+
+		for (auto &[depthstencil, image_view_data] : s_depth_stencil_buffer_imageViews)
+		{
+			if (image == image_view_data.image)
+			{
+				{ const std::lock_guard<std::mutex> lock(s_mutex);
+					// retrieve the current depth stencil image view
+					reshade::vulkan::draw_call_tracker *command_buffer_tracker = &s_trackers_per_command_buffer.at(commandBuffer);
+					if (image_view_data.image_data.image_info.usage & VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT)
+						track_cleared_depthstencil(commandBuffer, device, *command_buffer_tracker, depthstencil, image_view_data);
+					}
+			}
+		}
+	}
+
+	// important: the trampoline method must be executed after the depth buffer image saving
+	trampoline(commandBuffer, image, imageLayout, pDepthStencil, rangeCount, pRanges);
+}
+
+void     VKAPI_CALL vkCmdClearAttachments(VkCommandBuffer commandBuffer, uint32_t attachmentCount, const VkClearAttachment* pAttachments, uint32_t rectCount, const VkClearRect* pRects)
+{
+	PFN_vkCmdClearAttachments trampoline;
+
+	{ const std::lock_guard<std::mutex> lock(s_mutex);
+		trampoline = s_device_dispatch.at(get_dispatch_key(commandBuffer)).CmdClearAttachments;
+		assert(trampoline != nullptr);
+	}
+
+	// no device associated (this cannot happen normally)
+	if (const auto it = s_command_buffer_mapping.find(commandBuffer); it != s_command_buffer_mapping.end())
+	{
+		VkDevice device = s_command_buffer_mapping.at(commandBuffer);
+		VkFramebuffer framebuffer = s_framebuffer_per_command_buffer.at(commandBuffer);
+
+		const auto it2 = s_depth_stencil_buffer_frameBuffers.find(framebuffer);
+		if (it2 != s_depth_stencil_buffer_frameBuffers.end())
+		{
+			{ const std::lock_guard<std::mutex> lock(s_mutex);
+				// retrieve the current depth stencil image view
+				reshade::vulkan::draw_call_tracker *command_buffer_tracker = &s_trackers_per_command_buffer.at(commandBuffer);
+				image_view_data image_view_data = it2->second.image_view_data;
+				VkImageView depthstencil = it2->second.imageView;
+				if (image_view_data.image_data.image_info.usage & VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT)
+					track_cleared_depthstencil(commandBuffer, device, *command_buffer_tracker, depthstencil, image_view_data);
+			}
+		}
+	}
+
+	// important: the trampoline method must be executed after the depth buffer image saving
+	trampoline(commandBuffer, attachmentCount, pAttachments, rectCount, pRects);
+}
+
+VkResult VKAPI_CALL vkEndCommandBuffer(VkCommandBuffer commandBuffer)
```

Comment:
This is not doing anything from what I can see, so probably do not need to hook it.

Reply:
OK. Done.